### DISCUSSION TITLE
Issues with some classes having no effect

### DISCUSSION THREAD

**Discussion by jfschaff** [tailwindlabs/tailwindcss#11543] [Help] [answered]
Hi, I started a new project with Tailwind, and it's great! Thanks for the amazing work!

Unfortunately I have noticed issues with somes classes which seem to have no effect. I'm using v3.3.2 in a Node/React project.

Here is an example of something I don't understand:
```jsx
<div 
  id="navbar-sticky"
  className="items-center justify-between flex w-auto order-1"
>
  <ul 
    className="flex flex-row-reverse p-0 space-x-8 mt-0 border-0 justify-end"    // flex-row-reverse has no effect here
    style={{
      flexDirection: 'row-reverse',
    }}
  >
    <li className='flex items-center mx-2'>
      ...
    </li>
    <li className='flex items-center mx-2'>
      ...
    </li>
  </ul>
</div> 
```
In this example, when I remove the `style={{ flexDirection: 'row-reverse' }}` and inspect the page, I do see the class `flex-row-reverse` in the HTML, but not its corresponding style, and I don't have the expected `row-reverse` behavior. I need to use the style to get what I expect. This specific class is documented here: https://tailwindcss.com/docs/flex-direction#row-reversed

I also have that with a color in my navbar: a class `text-gray-400` in a span has no effect on color...

I also have that:
```jsx
<div className="flex items-center p-2"       // justify-end would have no effect here, I needed to use a style
  style={{
    justifyContent: 'end',
  }}
>
...
</div>
```

Otherwise my project makes extensive use of Tailwindcss and it works well!

Any idea what I'm missing?

**Comment by wongjn** [comment]
Consider checking:
- Tailwind has compiled since you added the class.
- The `content` file globs sufficiently cover the file that the code appears in.
- [String concatenation/interpolation is not used to build the class name](https://tailwindcss.com/docs/content-configuration#dynamic-class-names).

**Reply by jfschaff** [reply]
Thank you for your help!

> - Tailwind has compiled since you added the class.

I don't know how to check that, but restarting the dev server does not fix my issue.

> - The `content` file globs sufficiently cover the file that the code appears in.

Not sure what you mean, but other classes in the same source file do "work".

> - [String concatenation/interpolation is not used to build the class name](https://tailwindcss.com/docs/content-configuration#dynamic-class-names).

Thanks for the information, it's good to know. I'm not using concatenation or interpolation in the examples I gave.

**Reply by wongjn** [reply, marked as answer]
> > - The `content` file globs sufficiently cover the file that the code appears in.
>
> Not sure what you mean, but other classes in the same source file do "work".

Ensure the `content` file globs sufficiently cover the file that the code appears in. See [the documentation on content configuration](https://tailwindcss.com/docs/content-configuration) for full details.

**Reply by jfschaff** [reply]
You nailed it, thanks! I had not changed the `tailwind.config.js` file after switching my project from JS to TS and the `.tsx` extension was missing in the `content` section. For some reason, some classes were still working but not the "new ones" I added.

**Comment by mybearworld** [comment]
Have you overwritten these in your tailwindconfig anywhere?
